frontend/src/libs/hooks/use-debounce/use-debounce.hook.ts

```diff
@@ -0,0 +1,19 @@
+import { DEFAULT_DEBOUNCE_DELAY } from './libs/constants/constants.js';
+
+type DebounceFunction<T> = (...arguments_: T[]) => void;
+
+const useDebounce = <T>(
+  function_: DebounceFunction<T>,
+  delay?: number,
+): DebounceFunction<T> => {
+  let timeoutId: ReturnType<typeof setTimeout>;
+
+  return function (this: unknown, ...arguments_: T[]): void {
+    clearTimeout(timeoutId);
+    timeoutId = setTimeout(() => {
+      function_.apply(this, arguments_);
+    }, delay ?? DEFAULT_DEBOUNCE_DELAY);
+  };
+};
```

Comment:
Use 'debounce' library, no need to implement it yourself. It should be implemented as a function, not a hook. Also, it should implemented in shared folder. @Vika030718 should be working on it as well, please, contact her

Reply:
@fshabanov as i saw in Vika030718 pr she sends in search props setFilter func. so should i do the same thing and copy her debounce func?

Reply:
The Search component should receive a property onSearch, which should be wrapped by debounce inside of Search component.
I see the implementation of debounce is not fully ready, there are some comments, so do not copy exactly what she implemented, but take it into consideration, communicate with her about the implementation

Reply:
btw, do we actually need a debounce as a hook, if we want to wrap the function in it? Can't we just use a debounce as a helper function in this case?

Reply:
@fshabanov said it already

Reply:
done